frontend/src/pages/publish.jsx

```diff
@@ -291,23 +278,31 @@ const Publish = () => {
         setHasExistingContent(false);
       }
     };
-
     checkExistingContent();
   }, [selectedDomain, domain]);
 
   const publishOrFork = selectedDomain === domain ? 'Publish' : 'Fork';
-
   document.title = `${publishOrFork} - ${selectedDomain}`;
 
+  // --- minimal addition: format object diffs and hide legacy keys
+  const formatSettingsChange = (c) => {
+    const asObj = typeof c === 'object' && c !== null;
+    const path = asObj ? c.path : c;
+
+    // hide legacy entries from the Publish list
+    if (path === 'appearance.theme' || path === 'appearance.themeMode') return null;
+
+    if (!asObj) return String(c);
+    const from = JSON.stringify(c.from);
+    const to = JSON.stringify(c.to);
+    return `${c.path}: ${from} -> ${to}`;
+  };
+
```

Comment:
hm, formatting already happens within the settings.js class. Maybe this could just be an update to that method ([changeDiff()](https://github.com/stigmergic-org/simplepage/blob/main/packages/repo/src/settings.js#L176))?

Reply:
I think if we make the change it `changeDiff()` we shouldn't need any changes in the publish.jsx file at all?